src/fields/field.ts

```diff
@@ -33,11 +33,22 @@ export class BaseField {
     reconstructed = false,
   }: BaseFieldConstructor) {
     this.reconstructed = reconstructed;
-
-    if (valueKey in prediction && prediction[valueKey] !== null) {
+    if (
+      prediction !== undefined &&
+      prediction !== null &&
+      valueKey in prediction &&
+      prediction[valueKey] !== null
+    ) {
       this.value = prediction[valueKey];
     }
   }
+
+  toString(): string {
```

Comment:
it's a good idea to move this to the `BaseField` class, but you should also get rid of any identical methods that override it.

It looks like `Field` and `Amount` have identical overrides of it, so they should be removed.